### PR TITLE
add init container

### PR DESCRIPTION
**Pull request by pb82**
## JIRA ID

INTLY-3606

## Additional Information

Allow installing extensions through the Keycloak CR

## Verification Steps

1. Deploy keycloak or rhsso using the provided example CRs
2. Test the metrics endpoint under `<base url>/auth/realms/master/metrics`
3. Edit one of the Realm examples and add `eventsListeners: ["metrics-listener"]` to the spec
4. Create the realm and log in to it
5. Make sure the realm login event is recorded

## Checklist:
- [ ] Verified by team member
- [ ] Comments where necessary
- [ ] Automated Tests
- [ ] Documentation changes if necessary

### CODE REVIEW

File: pkg/model/constants.go [hunk 1]
```diff
@@ -2,26 +2,31 @@ package model
...
+	ApplicationName                     = "keycloak"
+	MonitoringKey                       = "middleware"
+	DatabaseSecretName                  = ApplicationName + "-db-secret"
+	PostgresqlPersistentVolumeName      = ApplicationName + "-postgresql-claim"
+	PostgresqlDeploymentName            = ApplicationName + "-postgresql"
+	PostgresqlDeploymentComponent       = "database"
+	PostgresqlServiceName               = ApplicationName + "-postgresql"
+	PostgresqlImage                     = "postgres:9.5"
+	KeycloakImage                       = "quay.io/keycloak/keycloak:7.0.1"
+	KeycloakInitContainerImage          = "quay.io/integreatly/sso_plugins_init:2.0.0"
```
Comment (davidffrench) [inline pkg/model/constants.go]: @pb82 How close are you to migrating the repo over? If it will be done tomorrow, would be good to switch this over.

Comment (pb82) [reply pkg/model/constants.go]: I'll update this on Monday

Comment (davidffrench) [reply pkg/model/constants.go]: Thanks @pb82

File: pkg/model/keycloak_deployment.go [hunk 2]
```diff
@@ -342,6 +314,65 @@ func KeycloakDeploymentReconciled(cr *v1alpha1.Keycloak, currentState *v13.State
 			},
 		},
 	}
-
+	reconciled.Spec.Template.Spec.InitContainers = KeycloakExtensionsInitContainers(cr)
 	return reconciled
 }
+
+func KeycloakExtensionsInitContainers(cr *v1alpha1.Keycloak) []v1.Container {
```
Comment (slaskawi) [inline pkg/model/keycloak_deployment.go]: These functions should probably be shared between Keycloak and RHSSO. Could you please make them unexported and put them in a separate file? I did similar thing in my backup PR and I used the following filename: `postgresql_common_backup.go`.

**Pull request merged**